src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -108,7 +103,13 @@ async ValueTask SynchronizeSolutionAssetsWorkerAsync()
             // the project states and we want to get that all in one batch.
             checksums.Clear();
             stateChecksums.AddAllTo(checksums);
-            await this.SynchronizeAssetsAsync(assetPath: AssetPath.SolutionAndTopLevelProjectsOnly, checksums, checksumToObjects, cancellationToken).ConfigureAwait(false);
+
+            using var _2 = PooledDictionary<Checksum, object>.GetInstance(out var checksumToObjects);
+
+            await this.SynchronizeAssetsAsync<object, Dictionary<Checksum, object>>(
+                assetPath: AssetPath.SolutionAndTopLevelProjectsOnly, checksums,
```

Comment:
>checksums, [](http://example.com/codeflow?start=70&length=10)

style preference: worth the extra line to not hide parameters when already multi-line